### PR TITLE
feat(can_share): can share chart and dashboard

### PR DESCRIPTION
**Pull request by amitmiran137**
### SUMMARY
As all things are so should sharing be a managed under a permission the user entitles to.
can_share_dashboard and can_share_chart are the new permissions .
both hide/show `ShareMenuItems` component on the dashboard/chart menus

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/47772523/114314495-a166ad00-9b03-11eb-9b10-604056ee85e8.mov

### TEST PLAN
1. check that sharing is menu option is shown by default for Admin/Gamma/Alpha
2. remove can share chart/dashboard on Superset from you user role
3. check that sharing menu option is hidden 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: tests/security_tests.py [hunk 1]
```diff
@@ -670,12 +670,13 @@ def assert_can_gamma(self, perm_set):
...
         self.assertIn(("can_dashboard", "Superset"), perm_set)
         self.assertIn(("can_explore", "Superset"), perm_set)
+        self.assertIn(("can_share_chart", "Superset"), perm_set)
+        self.assertIn(("can_share_dashboard", "Superset"), perm_set)
         self.assertIn(("can_explore_json", "Superset"), perm_set)
         self.assertIn(("can_fave_dashboards", "Superset"), perm_set)
         self.assertIn(("can_fave_slices", "Superset"), perm_set)
         self.assertIn(("can_save_dash", "Superset"), perm_set)
         self.assertIn(("can_slice", "Superset"), perm_set)
-        self.assertIn(("can_explore", "Superset"), perm_set)
```
Comment (etr2460) [inline tests/security_tests.py:678]: should this be removed?

**Review by etr2460 — COMMENTED** [review]
one question, otherwise i think this makes sense. I'll defer to @dpgaspar or someone more familiar with FAB permissions to stamp though

File: tests/security_tests.py (hunk 1 above)
Comment (amitmiran137) [reply tests/security_tests.py:678]: Yes bc it is duplicate row in row 674 so I just removed it

File: superset-frontend/src/dashboard/components/SliceHeaderControls/index.jsx [hunk 2]
```diff
@@ -253,18 +254,20 @@ class SliceHeaderControls extends React.PureComponent {
...
-            getActiveFilters(),
-            componentId,
-          )}
-          copyMenuItemTitle={t('Copy chart URL')}
-          emailMenuItemTitle={t('Share chart by email')}
-          emailSubject={t('Superset chart')}
-          addSuccessToast={addSuccessToast}
-          addDangerToast={addDangerToast}
-        />
+        {this.props.supersetCanShare && (
```
Comment (hughhhh) [inline superset-frontend/src/dashboard/components/SliceHeaderControls/index.jsx:257]: add this to line 183 decontruct `const` so you can just put `supersetCanShare`

**Review by dpgaspar — APPROVED** [review]
LGTM, 
Assuming that this is a frontend only permission stored on the server. We are not actually preventing user's from accessing the functionality just hiding it.

**Comment by graceguo-supercat** [comment]
This PR caused SHARE links (share url, share in email etc) are hidden in dashboard.

**Comment by amitmiran137** [comment]
If you run superset init I believe it will create by default the missing permission to Admin/Alpha/Gamma

**Comment by amitmiran137** [comment]
For backward compatibility you could create a migration that associate the permission to all existing roles

**Comment by junlincc** [comment]
@amitmiran137 was there a db migration made in this PR? 
look at the PR description, it was not super clear about all the permission changes that are being made. More details may help  prevent future confusion.

**Comment by amitmiran137** [comment]
No migration were need bc the PR sure that those 2 new permission are added by default once you run Superset init
And I as far as I know running superset init is a must command on every new deployment right after dB upgrade

**Pull request merged**